### PR TITLE
Add error messaging for new student sign-up page

### PR DESCRIPTION
**Pull request by TurnerRiley**
Adds error messaging for the new student sign-up page:
- If the `displayName` field is empty
- If the `age` field has no selection
- If the `state` field has no selection
- (if the parent checkbox is selected) If the `parentEmail` field is empty

### Page defaults to having the finish sign-up button be disabled
default screen

### Finish sign-up button is disabled if any/all errors are present
all error messages

### Finish sign-up button is enabled once no errors are present and required fields are filled in
button enabled once no errors

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-1990)
PR for error fields in the new teacher sign-up flow: [here](https://github.com/code-dot-org/code-dot-org/pull/60608)

## Testing story
Local testing and adding unit tests.

### CODE REVIEW

File: apps/src/signUpFlow/FinishStudentAccount.tsx [hunk 1]
```diff
@@ -165,6 +224,12 @@ const FinishStudentAccount: React.FunctionComponent<{
             iconStyle: 'solid',
             title: 'arrow-right',
           }}
+          disabled={
+            name === '' ||
+            age === '' ||
+            state === '' ||
+            (isParent && parentEmail === '')
+          }
```
Comment (bethanyaconnor) [inline apps/src/signUpFlow/FinishStudentAccount.tsx:232]: LOVE this! Will really cut down on the save button being pressed 🤩 

It seems like the only way to see the error messages is to set the fields and unset them, then, right?

File: apps/src/signUpFlow/FinishStudentAccount.tsx [hunk 2]
```diff
@@ -122,29 +160,50 @@ const FinishStudentAccount: React.FunctionComponent<{
...
+            onChange={onAgeChange}
+          />
+          {showAgeError && (
+            <BodyThreeText className={style.errorMessage}>
+              {locale.age_error_message()}
+            </BodyThreeText>
+          )}
+        </div>
+        <div>
+          <SimpleDropdown
```
Comment (bethanyaconnor) [inline apps/src/signUpFlow/FinishStudentAccount.tsx:193]: I know this isn't the change you're making in this PR but should the state dropdown be shown to all users? Or just US-based users?

**Review by bethanyaconnor — APPROVED** [review]

File: apps/src/signUpFlow/FinishStudentAccount.tsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/signUpFlow/FinishStudentAccount.tsx:232]: Yep!

File: apps/src/signUpFlow/FinishStudentAccount.tsx (hunk 2 above)
Comment (TurnerRiley) [reply apps/src/signUpFlow/FinishStudentAccount.tsx:193]: @dmcavoy Not sure? Currently there's a "I don't live anywhere listed here" option but that also seems a bit misleading for people who aren't in the U.S. altogether?

Comment (TurnerRiley) [reply apps/src/signUpFlow/FinishStudentAccount.tsx:193]: Happy to make a ticket for this!

Comment (dmcavoy) [reply apps/src/signUpFlow/FinishStudentAccount.tsx:193]: This is the student account page right? My understanding was that we only show the state dropdown to people we think are in the US. We may want to confirm with platform.

**Pull request merged**